### DISCUSSION TITLE
An unexpected error happened when adding the Apache IoTDB datasource in grafana.

### DISCUSSION THREAD

**Discussion by rsupreet** [apache/iotdb#9758] [Q&A] [answered]
**OS**: 
Linux CentOS 7

**Software versions installed**:
Apache IoTDB - [All-in-one] - 1.1.0
Apache IoTDB - [Grafana-plugin] - 1.1.0
Grafana - OSS - 7.5.17

Apache IoTDB Grafana plugin v1.1.0 is not working with the Grafana OSS v7.5.17, when tried to add the Apache IoTDB data source in Grafana the following exception occurred without displaying the data source configuration settings screen:

```
An unexpected error happened
DetailsError: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

    at div
    at mv (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:2048205)
    at div
    at a (http://<IP_ADDRESS:PORT>/public/plugins/apache-iotdb-datasource/module.js:2:21789)
    at div
    at t (http://<IP_ADDRESS:PORT>/public/build/DataSourceSettingsPage.28bc2c784e75f5550d3e.js:1:4239)
    at form
    at div
    at div
    at ZGyg.m.Contents (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:1728533)
    at div
    at div
    at div
    at t (http://<IP_ADDRESS:PORT>/public/build/vendors~app.28bc2c784e75f5550d3e.js:2:1105455)
    at m (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:3052725)
    at div
    at m (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:1728033)
    at t (http://<IP_ADDRESS:PORT>/public/build/DataSourceSettingsPage.28bc2c784e75f5550d3e.js:1:8963)
    at O (http://<IP_ADDRESS:PORT>/public/build/vendors~app.28bc2c784e75f5550d3e.js:2:90047)
    at l (http://<IP_ADDRESS:PORT>/public/build/DataSourceSettingsPage.28bc2c784e75f5550d3e.js:1:14478)
    at n (http://<IP_ADDRESS:PORT>/public/build/vendors~app.28bc2c784e75f5550d3e.js:2:731856)
    at t (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:1801236)
    at h (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:2976416)
    at t (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:1963025)
    at t (http://<IP_ADDRESS:PORT>/public/build/app.28bc2c784e75f5550d3e.js:2:1963555)
    at l (http://<IP_ADDRESS:PORT>/public/build/vendors~app.28bc2c784e75f5550d3e.js:2:87654)

```
Does the Apache IoTDB v1.1.0 supports Grafana OSS v7.5.17 ? 
If not supported then what is the alternate for using Grafana with Apache IoTDB?

**Comment by SteveYurongSu** [comment]
Hi @rsupreet, thanks for reporting this.

Does the Apache IoTDB v1.1.0 supports Grafana OSS v7.5.17 ? @CloudWise-Lukemiao 

Please have a try with the Grafana OSS >= v9.3.0 @rsupreet

**Reply by rsupreet** [reply]
Hi @SteveYurongSu,

Thank you for the reply.

> Please have a try with the Grafana OSS >= v9.3.0 @rsupreet

I tried testing of Apache IoTDB Grafana plugin v1.1.0 with Grafana OSS >=9.3.0, is working fine and I am able to add the data source without any errors.

But I am using **Grafana OSS v7.5.17** version (having Apache 2.0 license) in my application and _I want to integrate latest stable version of Apache IoTDB with Grafana OSS v7.5.17_.
Request you to please guide me on how I can make progress with this?

**Reply by rsupreet** [reply]
Hi @SteveYurongSu, @CloudWise-Lukemiao,

Request you to please update me on **_how to integrate the recent stable version of Apache IoTDB with the Grafana OSS version 7.x.x(Eg: 7.5.17)_**?
In latest version of Apache IoTDB documentation(https://iotdb.apache.org/UserGuide/Master/Ecosystem-Integration/Grafana-Plugin.html#installation-and-deployment) it is mentioned as Apache IoTDB supports Grafana Version >= 7.0.0, but I get the unexpected error(as mentioned above) while adding the recent version of Apache IoTDB data source in Grafana v7.5.17.

**Reply by CloudWise-Lukemiao** [reply]
> Hi @SteveYurongSu, @CloudWise-Lukemiao,
> 
> Request you to please update me on **_how to integrate the recent stable version of Apache IoTDB with the Grafana OSS version 7.x.x(Eg: 7.5.17)_**? In latest version of Apache IoTDB documentation(https://iotdb.apache.org/UserGuide/Master/Ecosystem-Integration/Grafana-Plugin.html#installation-and-deployment) it is mentioned as Apache IoTDB supports Grafana Version >= 7.0.0, but I get the unexpected error(as mentioned above) while adding the recent version of Apache IoTDB data source in Grafana v7.5.17.

hi @rsupreet ,
I found from the above information that the IoTDB v1.0.0 version is used, but the Grafana plugin uses v1.1.0, which makes it impossible to obtain data. So if you are using iotdb v 1.0.0, please download the same version of (https://archive.apache.org/dist/iotdb/1.0.0/) from here, and please use v9.3 and above for grafana.
Finally, thank you for pointing out the issue with the latest version of the document. I will fix it.

**Reply by rsupreet** [reply]
Hi @CloudWise-Lukemiao,

> hi @rsupreet ,
> I found from the above information that the IoTDB v1.0.0 version is used, but the Grafana plugin uses v1.1.0, which makes it impossible to obtain data. So if you are using iotdb v 1.0.0, please download the same version of (https://archive.apache.org/dist/iotdb/1.0.0/) from here, and please use v9.3 and above for grafana.
> Finally, thank you for pointing out the issue with the latest version of the document. I will fix it.

As you said, I  am using similar versions of IoTDB and Grafana plugin(i.e. **_IoTDB v1.1.0_** and **_Grafana plugin v1.1.0_**), but using **_Grafana v7.5.17_**. 
Will this combination work or updating to Grafana version >= 9.3.0 is mandatory for Apache IoTDB v1.1.0?

**Reply by SteveYurongSu** [reply, marked as answer]
> Hi @CloudWise-Lukemiao,
> 
> > hi @rsupreet ,
> > I found from the above information that the IoTDB v1.0.0 version is used, but the Grafana plugin uses v1.1.0, which makes it impossible to obtain data. So if you are using iotdb v 1.0.0, please download the same version of (https://archive.apache.org/dist/iotdb/1.0.0/) from here, and please use v9.3 and above for grafana.
> > Finally, thank you for pointing out the issue with the latest version of the document. I will fix it.
> 
> As you said, I am using similar versions of IoTDB and Grafana plugin(i.e. **_IoTDB v1.1.0_** and **_Grafana plugin v1.1.0_**), but using **_Grafana v7.5.17_**. Will this combination work or updating to Grafana version >= 9.3.0 is mandatory for Apache IoTDB v1.1.0?

@rsupreet Sorry for the late reply. 

The combination (IoTDB v1.1.0 + Grafana plugin v1.1.0 + Grafana v7.5.17) won't work. Updating to Grafana version >= 9.3.0 is mandatory for Apache IoTDB v1.1.0. :(

**Reply by rsupreet** [reply]
Hi @SteveYurongSu,



> > Hi @CloudWise-Lukemiao,
> > > hi @rsupreet ,
> > > I found from the above information that the IoTDB v1.0.0 version is used, but the Grafana plugin uses v1.1.0, which makes it impossible to obtain data. So if you are using iotdb v 1.0.0, please download the same version of (https://archive.apache.org/dist/iotdb/1.0.0/) from here, and please use v9.3 and above for grafana.
> > > Finally, thank you for pointing out the issue with the latest version of the document. I will fix it.
> > 
> > 
> > As you said, I am using similar versions of IoTDB and Grafana plugin(i.e. **_IoTDB v1.1.0_** and **_Grafana plugin v1.1.0_**), but using **_Grafana v7.5.17_**. Will this combination work or updating to Grafana version >= 9.3.0 is mandatory for Apache IoTDB v1.1.0?
> 
> @rsupreet Sorry for the late reply.
> 
> The combination (IoTDB v1.1.0 + Grafana plugin v1.1.0 + Grafana v7.5.17) won't work. Updating to Grafana version >= 9.3.0 is mandatory for Apache IoTDB v1.1.0. :(

Thank you for the confirmation.
In future releases of Apache IoTDB will there be a support to Grafana OSS 7.x.x versions (having Apache 2.0 license)?

**Reply by CloudWise-Lukemiao** [reply]
Due to the use of new controls provided by Grafana in the latest Grafana plugin, it is currently incompatible with versions below Grafana 9.3. If you want to use the version of Grafana 7, you can use Iotdb1.1 and Grafana plugin version 0.13.3（ https://archive.apache.org/dist/iotdb/0.13.3/ )。

**Reply by SteveYurongSu** [reply]
@CloudWise-Lukemiao Thanks for providing this method. 

@rsupreet Grafana plugin version 0.13.3 can also be a compatible version (it uses the compatible version of openapi connecting IoTDB REST server).

**Reply by rsupreet** [reply]
Hi @CloudWise-Lukemiao, @SteveYurongSu,

Thank you for the support and confirmation. Will try to integrate with the compatible versions you mentioned above.
